File: sotetsuk/ALDS1_1_D.py
n = int(raw_input())
R_list = [int(raw_input()) for _ in range(n)]

min_R = int(1e10)
ans = -int(1e10)
for R in R_list:
    ans = max(ans, R - min_R)
    min_R = min(min_R, R)

print ans

File: sotetsuk/AOJ/src/ALDS1_1_D.scala
import java.util.Scanner

object ALDS1_1_D { // -> Main
  def main(args: Array[String]) {
    val sc = new Scanner(System.in)
    val n = sc.nextInt
    val list = List.fill(n)(sc.nextInt)

    var minr = 1e10.toInt
    var ans = -1e10.toInt
    for (r <- list) {
      ans = math.max(ans, r - minr)
      minr = math.min(minr, r)
    }
    println(ans)
  }
}